06_puppy/jimbotech/sync_mapstore.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+// SyncMapStore stores puppies threadsafe.
+type SyncMapStore struct {
+	sync.Map
+}
+
+// length is not concorrency safe. As the go doc says:
+// Range does not necessarily correspond to any consistent snapshot of the
+// Map's contents: no key will be visited more than once, but if the value
+// for any key is stored or deleted concurrently, Range may reflect any
+// mapping for that key from any point during the Range call.
+//
+func (s *SyncMapStore) length() int {
+	var length int
+	s.Range(func(key interface{}, value interface{}) bool {
+		length++
+		return true
+	})
+	return length
+}
+
+// CreatePuppy threadsafe adding a puppy to storage
+// but will modify the member ID.
+func (s *SyncMapStore) CreatePuppy(p *Puppy) (uint32, error) {
+	p.ID = uuid.New().ID()
+	sp := p
```

Comment:
This is not making a copy of the Puppy - you're just assigning the pointer. You fixed this in mapStorem but not here by the looks of it.

Maybe you should add test cases to ensure that the caller cannot change puppy in the store through either a pointer passed to the store or a pointer returned from the store.